src/main/java/org/assertj/assertions/generator/BaseAssertionGenerator.java

```diff
@@ -362,12 +401,20 @@ private String generateAssertionEntryPointMethodsFor(final Set<ClassDescription>
       String assertionEntryPointMethodContent = assertionEntryPointMethodTemplate.getContent();
       // resolve class assert (ex: PlayerAssert)
       // in case of inner classes like Movie.PublicCategory, class assert will be MoviePublicCategoryAssert
+      String customAssertionClass = fullyQualifiedAssertClassName(classDescription);
+      if (imports.contains(customAssertionClass)) {
```

Comment:
I was thinking to recompute the imports here only to be able to decide whether to use a FQN or not, replacing `${imports}` would still be done in `generateAssertionsEntryPointClassContent`.
I prefer to have a simpler method signature (without `Set<String> imports` parameter) at the cost of recompute the set of imports which is not an overly expensive operation, I usually don't like to pass parameters when they can be inferred in the method itself.